lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3008,8 +3008,29 @@ void AttributeChecker::visitPropertyWrapperAttr(PropertyWrapperAttr *attr) {
 }
 
 void AttributeChecker::visitFunctionBuilderAttr(FunctionBuilderAttr *attr) {
-  // TODO: check that the type at least provides a `sequence` factory?
-  // Any other validation?
+  auto *nominal = dyn_cast<NominalTypeDecl>(D);
+  auto lookupResult = TypeChecker::lookupMember(
+      nominal, nominal->getDeclaredType(),
+      DeclNameRef(D->getASTContext().Id_buildBlock));
```

Comment:
I think you need to pass an empty identifier as well, as `buildBlock` takes a single parameter with no argument label.

Reply:
I think the following should be valid, if weird:
```
@_functionBuilder
struct Builder {
    static func buildBlock() -> Int { return 3 }
}

class C {
    @Builder func foo() -> Int {}
}
```